Civi/Api4/Service/Spec/Provider/GroupGetSpecProvider.php

```diff
@@ -60,4 +69,17 @@ public static function countContacts(array $field): string {
     )";
   }
 
+  /**
+   * Generate SQL for checking cache expiration for smart groups and parent groups
+   *
+   * @return string
+   */
+  public static function getCacheExpiredSQL(array $field): string {
+    $smartGroupCacheTimeoutDateTime = \CRM_Contact_BAO_GroupContactCache::getCacheInvalidDateTime();
+    $cacheDate = $field['sql_name'];
+    $savedSearchId = substr_replace($field['sql_name'], 'saved_search_id', -11, -1);
+    $children = substr_replace($field['sql_name'], 'children', -11, -1);
```

Comment:
I had to read that a couple of times to figure out what you're doing!  Can we make the prefix available separately?